lib/rubocop/cop/rspec/let_setup.rb

```diff
@@ -45,10 +51,19 @@ def on_block(node)
         private
 
         def unused_let_bang(node)
-          let_bang(node) do |method_send, method_name|
+          child_let_bang(node) do |method_send, method_name|
             yield(method_send) unless method_called?(node, method_name)
           end
         end
+
+        def child_let_bang(node)
+          lets = RuboCop::RSpec::ExampleGroup.new(node).lets
+
+          lets.each do |let|
+            node_with_name = let_bang(let)
+            yield node_with_name if node_with_name
+          end
```

Comment:
Maybe this has been discussed already, but this could be rewritten to get rid of the `nil` check:
```rb
lets.each do |let|
  let_bang(let) do |send_node, method_name|
    yield send_node, method_name
  end
end
```

Reply:
Well noticed 👍 
Maybe even
```ruby
lets.each do |let|
  let_bang(let, &block)
end
```

Reply:
I don't think so. `RuboCop::RSpec::ExampleGroup#lets` returns both `let` and `let!` but here we need only `let`s with bang. That's why there is the `nil` check.

Reply:
yeah but `let_bang` would not yield if the node is not matched ;) So there is a difference between passing a block to a node_matcher or using the matcher's return value (which might be nil)

Reply:
Right, but we filter this through `let_bang` that yields only when there is something to yield (bang-let found).
Checked both options, specs pass.

My final version of the method, leaving it up to you which option to pick:
```ruby
        def child_let_bang(node, &block)
          RuboCop::RSpec::ExampleGroup.new(node).lets
            .each { |let| let_bang(let, &block) }
        end
```

Reply:
Great. Will update the PR soon.

Reply:
Done